bukkit/src/main/java/com/github/games647/fastlogin/bukkit/listener/protocollib/VerifyResponseTask.java

```diff
@@ -175,13 +176,31 @@ private boolean enableEncryption(SecretKey loginKey) throws IllegalArgumentExcep
             //get the NMS connection handle of this player
             Object networkManager = getNetworkManager();
 
-            //try to detect the method by parameters
-            Method encryptMethod = FuzzyReflection
-                    .fromObject(networkManager).getMethodByParameters("a", SecretKey.class);
-
-            //encrypt/decrypt following packets
-            //the client expects this behaviour
-            encryptMethod.invoke(networkManager, loginKey);
+            try {
+                //try to detect the method by parameters
+                Method encryptMethod = FuzzyReflection
+                        .fromObject(networkManager).getMethodByParameters("a", SecretKey.class);
+
+                //encrypt/decrypt following packets
+                //the client expects this behaviour
+                encryptMethod.invoke(networkManager, loginKey);
+            } catch (Exception ex) {
+                //try to detect the method by parameters
+                Method encryptMethod = FuzzyReflection
+                        .fromObject(networkManager).getMethodByParameters("a", Cipher.class, Cipher.class);
+
+                //change Key instance into two Cipher instances
+                String ver = Bukkit.getServer().getClass().getPackage().getName().split("\\.")[3];
+                Class<?> encryptionClass = Class.forName("net.minecraft.server." + ver + ".MinecraftEncryption");
```

Comment:
ProtocolLib has a helper for finding the NMS class. Then we could rely on the unit tests provided by ProtocolLib.

Reply:
I've switched to using MinecraftReflection class